trace/tr_test.go

```diff
@@ -0,0 +1,150 @@
+package trace_test
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"net/http"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/TyeMcQueen/go-lager"
+	"github.com/TyeMcQueen/go-lager/gcp-spans"
+	"github.com/TyeMcQueen/go-tutl"
+	"github.com/TyeMcQueen/tools-gcp/trace"
+)
+
+func TestTrace(t *testing.T) {
+	u := tutl.New(t)
+	logs := &bytes.Buffer{}
+	defer u.Is("", logs.Bytes(), "no final errors")
+	defer lager.SetOutput(logs)()
+
+	proj := os.Getenv("GCP_PROJECT_ID")
+	if "" == os.Getenv("GCP_PROJECT_ID") {
+		out, err := os.OpenFile("/dev/tty", os.O_WRONLY, 0666)
+		if nil != err {
+			u.Log("Can't write to /dev/tty: ", err)
+			out = os.Stderr
+		}
+		fmt.Fprintf(out, "SKIPPED: %s.\n",
+			"Testing requires GCP_PROJECT_ID set and default creds")
+		t.SkipNow()
+		return
+	}
+
+	ctx := context.Background()
+	var spanReg *trace.Registrar
+	defer trace.StartServer(&ctx, 1, &spanReg)()
+	empty := spans.ContextGetSpan(ctx)
+	u.IsNot(nil, empty, "NewFactory")
+
+	u.Is(proj, empty.GetProjectID(), "empty GetProjectID")
```

Comment:
since it's not strictly empty, should this be called something else?

Reply:
Each Factory either contains a span or is called "empty".  But even an empty factory still needs to be able to function as a factory.  So the object still represents a factory, just one with no span in stock.  If it didn't have the machinery, it wouldn't be a factory, it'd be an empty object that can't work as a factory.  Though this is clearer in the non-read-only factory case.

Reply:
Okay cool - the "empty" refers to the factory rather than the Project ID - got it